src/test/java/com/sdase/k8s/operator/mongodb/controller/tasks/util/ConnectionStringUtilTest.java

```diff
@@ -89,4 +89,23 @@ void shouldIgnoreOptionsIfOptionsAreNotSet() {
                 + ",some-documentdb.c789012.eu-central-1.docdb.amazonaws.com:27017"
                 + "/my-service-db");
   }
+
+  @Test
+  void shouldSupportMongoDbSrvSchema() {
+    var given = "mongodb+srv://server.example.com/admin?connectTimeoutMS=2000";
+    var username = "my-service-user";
+    var password = "dummy";
+    var database = "my-service-db";
+    var connectionString = new ConnectionString(given);
+    var actual =
+        ConnectionStringUtil.createConnectionString(
+            database, username, password, null, connectionString);
```

Comment:
Not really clear if the standalone properties are actually used here or the full connection string. Could you clarify?

Reply:
The test is just a copy from the one above with different schema. In general the `ConnectionStringUtil` takes the connection string used by the operator itself as input and replaces username, data, password and optional custom options with the information needed for the specific requested database requested by a deployment. So in the end it is expected that you get the input connection string but username, password and database replaced by the individual properties.

Reply:
I see. When looking up the documentation I also get:
> Overwrites a MongoDB {@linkplain ConnectionString} with the passed arguments.

Was confused because the approach in sda-dropwizard-commons was that the connectionString always wins. Did not see the necessity to replace properties.